src/main/java/minijava/ir/optimize/JmpBlockRemover.java

```diff
@@ -26,33 +31,47 @@ public boolean optimize(Graph graph) {
 
   @Override
   public void visit(Block block) {
-    if (isJmpBlock(block) && hasOnePredecessor(block)) {
-      hasChanged = true;
-      remove(block);
-    }
+    tryToIdentifyNewEdge(block)
+        .ifPresent(
+            edge -> {
+              hasChanged = true;
+              remove(block, edge.v1, edge.v2, edge.v3);
+            });
   }
 
-  private boolean isJmpBlock(Block block) {
+  private Optional<Tuple3<Node, Integer, Block>> tryToIdentifyNewEdge(Block block) {
```

Comment:
There's nothing functional about tuples, but a separate type instead of a meaningless tuple would be great I guess.

Otherwise you have to pretty much duplicate everything in that method.